libgnucash/app-utils/date-utilities.scm

```diff
@@ -430,8 +430,12 @@
 (define (gnc:get-absolute-from-relative-date date-symbol)
   (let ((rel-date-data (hash-ref gnc:relative-date-hash date-symbol)))
     (if rel-date-data
-       ((gnc:reldate-get-fn rel-date-data))
-       (gnc:error "Tried to look up an undefined date symbol"))))
+        ((gnc:reldate-get-fn rel-date-data))
+        (let ((msg (format #f "Tried to look up an undefined date symbol '~a'. \
+This report was probably saved by a later version of GnuCash. Defaulting \
+to today." date-symbol)))
+          (gnc:gui-warn msg)
+          (current-time)))))
```

Comment:
The warning is no longer printed to the trace file as suggested by the commit message. Was that deliberate ?

Reply:
It should... in the redefinition of `gnc:gui-warn` does call the old `gnc:warn` after clicking Close... perhaps I'll reverse the order...

Reply:
Ah, I missed that call. I see now.

However that turns the issue around: somewhere below you call  rpterror_earlier followed by a gnc:error. As the former calls gnc:warn you now send two messages to the trace log. One being a warning and the other being an error. That's confusing unfortunately.

Also in the light of the translations, be careful to send the English text to the log and present the translated version to the user...

Reply:
> However that turns the issue around: somewhere below you call rpterror_earlier followed by a gnc:error. As the former calls gnc:warn you now send two messages to the trace log. One being a warning and the other being an error. That's confusing unfortunately.

Ok found the double log. No difficulty removing old one.

> Also in the light of the translations, be careful to send the English text to the log and present the translated version to the user...

Ack, this one *not* easy! The formatted string gets sent to both gnc:warn and gui-warning-dialog!! If this is important to resolve perhaps *every* gnc:gui-error/warn/msg must send two strings - one for UI and the other for console. 
```
(define (rpterror-earlier type newoption fallback)
  (let ((template (N_ "This report was saved using a later version of \
GnuCash. One of the newer ~a options '~a' is not available, fallback to the \
option '~a'."))
        (console-msg (format #f template type newoption fallback))
        (ui-msg (format #f (_ template) type newoption fallback)))
    (gnc:gui-warn console-msg ui-msg)))
```

Is this desirable?

Reply:
I've created last amendment for translations but think they add unnecessary complexity. Every gnc:gui-error will need to generate two strings.

cf price-quotes.scm also generates either `gnc-error-dialog` or `gnc:warn` and uses translated strings only. I'm not sure if price-quotes.scm ever gets to run without UI that's beside the point.

Reply:
Yes, price-quotes.scm is run from the command-line via the --get-price-quotes option. Many users put it in a cron job/scheduled task to get quotes at the end of each trading day.

Your approach declaring the string in (N_ ...) and calling (_ ...) for dispatch to the gui is correct.

Reply:
Ok, I've fixed up the gui functions and the crasher. Await travis, then merge!